assets/js/base/context/cart-checkout/payment-methods/use-payment-method-registration.js

```diff
@@ -86,8 +89,13 @@ const usePaymentMethodRegistration = (
 				[ paymentMethod.name ]: paymentMethod,
 			};
 		};
-		for ( const paymentMethodName in registeredPaymentMethods ) {
+
+		for ( let i=0; i<PAYMENT_GATEWAY_ORDER.length; i++  ) {
```

Comment:
If we managed to get the sorting value in the registered payment methods config we could pre-sort these using something like https://lodash.com/docs/4.17.15#sortBy and avoid changing these loops.